### PR TITLE
Refactor method and variable names to pep8_style

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #19

### CODE REVIEW

**Comment by MichaelClerx** [comment]
For some reason no tests are being run on this branch...?

**Comment by MichaelClerx** [comment]
> For some reason no tests are being run on this branch...?

That's because it was set to merge into a different branch than master

File: fc/simulations/modifiers.py [hunk 1]
```diff
@@ -44,46 +44,47 @@ def __getstate__(self):
...
 
         We cache most of the lookups required locally, since the simulation object we are passed will be
         the same each time.
         """
-        self._bindings[self.varLocalName] = self._evaluate(simul.env)
+        self._bindings[self.var_local_name] = self._evaluate(simul.env)
 
 
-class SaveState(AbstractModifier):
+class save_state(AbstractModifier):
```
Comment (jonc125) [inline fc/simulations/modifiers.py]: Class not function

File: fc/simulations/modifiers.py [hunk 2]
```diff
@@ -44,46 +44,47 @@ def __getstate__(self):
...
+        self.state_name = state_name
 
-    def Apply(self, simul):
-        simul.model.SaveState(self.stateName)
+    def apply(self, simul):
+        simul.model.save_state(self.state_name)
 
 
-class ResetState(AbstractModifier):
+class reset_state(AbstractModifier):
```
Comment (jonc125) [inline fc/simulations/modifiers.py]: Class not function

File: fc/simulations/ranges.py [hunk 3]
```diff
@@ -9,24 +9,24 @@ class AbstractRange(V.Simple):
     """
 
     def __init__(self, name):
-        """Initialise the common range properties."""
+        """initialise the common range properties."""
```
Comment (jonc125) [inline fc/simulations/ranges.py]: No

File: fc/simulations/ranges.py [hunk 4]
```diff
@@ -117,29 +117,29 @@ def __iter__(self):
...
+            self.value = self.arr_range[self.count]
             return self.current
 
 
 class While(AbstractRange):
     def __init__(self, name, condition):
         super(While, self).__init__(name)
         self.condition = condition
-        self._Init()
+        self.__init()
```
Comment (jonc125) [inline fc/simulations/ranges.py]: Why double not single underscore?

File: fc/simulations/simulations.py [hunk 5]
```diff
@@ -53,141 +53,141 @@ def __getstate__(self):
...
             # Note that the tuple conversion in the next line is very quick
             range_indices = tuple(map(attrgetter('count'), self.ranges))  # tuple(r.count for r in self.ranges)
-            for output, result in zip(outputsList, results_list):
+            for output, result in zip(outputs_list, results_list):
                 result[range_indices] = output
 
 
 class Timecourse(AbstractSimulation):
-    """Simulate a simple loop over time."""
+    """simulate a simple loop over time."""
```
Comment (jonc125) [inline fc/simulations/simulations.py]: No

File: fc/simulations/simulations.py [hunk 6]
```diff
@@ -219,77 +219,77 @@ def InternalRun(self, verbose=True):
                 start_hook()
             add_outputs(get_outputs())
             end_hook()
-        self.LoopEndHook()
+        self.loop_end_hook()
 
 
 class OneStep(AbstractSimulation):
-    """Simulate one logical execution of a model."""
+    """simulate one logical execution of a model."""
```
Comment (jonc125) [inline fc/simulations/simulations.py]: No

File: fc/simulations/simulations.py [hunk 7]
```diff
@@ -219,77 +219,77 @@ def InternalRun(self, verbose=True):
...
 
-    class NullRange(R.AbstractRange):
+    class null_range(R.AbstractRange):
```
Comment (jonc125) [inline fc/simulations/simulations.py]: Class not function

File: fc/sundials/solver.pxd [hunk 8]
```diff
@@ -10,9 +10,9 @@ IF FC_SUNDIALS_MAJOR >= 3:
     ctypedef _lib.SUNLinearSolver SUNLinearSolver
 
 
-#cdef object NumpyView(N_Vector v)
+#cdef object numpy_view(N_Vector v)
```
Comment (jonc125) [inline fc/sundials/solver.pxd]: Maybe get rid of these if they're commented out and never used?

File: fc/test_support.py [hunk 9]
```diff
@@ -119,67 +121,68 @@ def CheckResults(proto, expectedSpec, dataFolder, rtol=0.01, atol=0, messages=No
...
         real_path += '.gz'
         is_compressed = True
     return real_path, is_compressed
 
 
-def Load2d(filePath):
-    """Load the legacy data format for 2d arrays."""
-    real_path, is_compressed = CheckFileCompression(filePath)
+def load2d(file_path):
+    """load the legacy data format for 2d arrays."""
```
Comment (jonc125) [inline fc/test_support.py]: No

File: fc/language/expressions/abstract.py [hunk 10]
```diff
@@ -107,63 +107,64 @@ def evalGlobals(self):
             d['___np'] = numpy
             return d
 
-    def RealEvaluateCompiled(self, env):
-        """Evaluate the compiled form of this expression using eval().
+    def real_evaluate_compiled(self, env):
+        """evaluate the compiled form of this expression using eval().
```
[the remark this answers is not in the document]
Comment (jonc125) [reply fc/language/expressions/abstract.py]: As in, just the occurrence in the comment.

**Comment by MichaelClerx** [comment]
Thanks. Will have a look at this on Friday!

File: test/test_model_simulations.py [hunk 11]
```diff
@@ -9,170 +9,170 @@
...
+        np.testing.assert_array_almost_equal(predicted, results.look_up('y').array)
 
-    def testSetWithRange(self):
+    def test_set_with_range(self):
         a = 5
         model = Model.TestOdeModel(a)
-        set_modifier = Modifiers.SetVariable(Modifiers.AbstractModifier.START_ONLY,
-                                             'oxmeta:leakage_current', E.NameLookUp('count'))
-        reset_modifier = Modifiers.ResetState(Modifiers.AbstractModifier.START_ONLY)
+        set_modifier = Modifiers.set_variable(
```
Comment (jonc125) [inline test/test_model_simulations.py]: SetVariable class again

File: test/test_nd_array.py [hunk 12]
```diff
@@ -6,15 +6,15 @@
 class TestNdArray(unittest.TestCase):
     """Test functionality of numpy arrays. No protocol arrays tested here."""
 
-    def test0dArray(self):
+    def test0d_array(self):
```
Comment (jonc125) [inline test/test_nd_array.py]: test_0d ?

File: test/test_python_reproducibility.py [hunk 13]
```diff
@@ -46,15 +46,15 @@ def __exit__(self, exc_type, exc_value, traceback):
...
     """Function run on initialization of a Pool worker process.
 
     If the setproctitle module is installed, this will adjust the process title (shown by ps) to be more informative.
     """
-    setproctitle('python worker %d' % test_support.GetProcessNumber())
+    setproctitle('python worker %d' % test_support.get_process_number())
 
 
-def RunExperiment(modelName, protoName, expectedOutputs):
+def run_experiment(modelName, protoName, expectedOutputs):
```
Comment (jonc125) [inline test/test_python_reproducibility.py]: Haven't changed arg names. But this isn't testable at present so probably best not to touch this file yet?

File: test/test_syntax_interface.py [hunk 14]
```diff
@@ -10,751 +10,749 @@
...
+        self.assertEqual(expr.evaluate(env).value, 8.0)
 
-        parse_action = csp.expr.parseString('2.0 ^ 3.0 == 5.0 + 3.0', parseAll=True)
+        parse_action = CSP.expr.parseString('2.0 ^ 3.0 == 5.0 + 3.0', parseAll=True)
         expr = parse_action[0].expr()
-        self.assertEqual(expr.Evaluate(env).value, 1)
+        self.assertEqual(expr.evaluate(env).value, 1)
 
-    def testParsingMathMLFuncs(self):
+    def test_parsing_math_m_l_funcs(self):
```
Comment (jonc125) [inline test/test_syntax_interface.py]: MathML is a name, keep as is

File: test/test_syntax_interface.py [hunk 15]
```diff
@@ -10,751 +10,749 @@
...
         expr = fold_parse_action[0].expr()
-        result = expr.Evaluate(env)
+        result = expr.evaluate(env)
         predicted = np.array([[10], [16]])
         np.testing.assert_array_almost_equal(result.array, predicted)
 
-    def testProtocolandPostProcessing(self):
-        env = Env.Environment()
-        parse_action = csp.postProcessing.parseString('post-processing{a=2}')
+    def test_protocoland_post_processing(self):
```
Comment (jonc125) [inline test/test_syntax_interface.py]: protocol_and

File: test/test_syntax_interface.py [hunk 16]
```diff
@@ -10,751 +10,749 @@
...
         expr = parse_action[0].expr()
         self.assertEqual(expr, Modifiers.AbstractModifier.END_ONLY)
 
-        parse_action = csp.modifier.parseString('at start set model:a = 5.0', parseAll=True)
+        parse_action = CSP.modifier.parseString('at start set model:a = 5.0', parseAll=True)
         expr = parse_action[0].expr()
-        self.assertIsInstance(expr, Modifiers.SetVariable)
-        self.assertEqual(expr.variableName, 'model:a')
-        self.assertEqual(expr.valueExpr.value.value, 5)
+        self.assertIsInstance(expr, Modifiers.set_variable)
```
Comment (jonc125) [inline test/test_syntax_interface.py]: class

File: test/test_syntax_interface.py [hunk 17]
```diff
@@ -10,751 +10,749 @@
...
+        self.assertEqual(expr.variable_name, 'model:a')
+        self.assertEqual(expr.value_expr.value.value, 5)
 
-        parse_action = csp.modifier.parseString('at start set model:t = 10.0', parseAll=True)
+        parse_action = CSP.modifier.parseString('at start set model:t = 10.0', parseAll=True)
         expr = parse_action[0].expr()
-        self.assertIsInstance(expr, Modifiers.SetVariable)
-        self.assertEqual(expr.variableName, 'model:t')
-        self.assertEqual(expr.valueExpr.value.value, 10)
+        self.assertIsInstance(expr, Modifiers.set_variable)
```
Comment (jonc125) [inline test/test_syntax_interface.py]: class

File: test/test_syntax_interface.py [hunk 18]
```diff
@@ -10,751 +10,749 @@
...
+        self.assertEqual(expr.variable_name, 'model:t')
+        self.assertEqual(expr.value_expr.value.value, 10)
 
-        parse_action = csp.modifier.parseString('at start save as state_name', parseAll=True)
+        parse_action = CSP.modifier.parseString('at start save as state_name', parseAll=True)
         expr = parse_action[0].expr()
-        self.assertIsInstance(expr, Modifiers.SaveState)
-        self.assertEqual(expr.stateName, 'state_name')
+        self.assertIsInstance(expr, Modifiers.save_state)
```
Comment (jonc125) [inline test/test_syntax_interface.py]: class

File: test/ContinuousTestPack.txt [hunk 19]
```diff
@@ -1,14 +1,14 @@
...
-TestFunctions.py
-TestLogicOperators.py
-TestMathmlEvaluation.py
-TestModelSimulations.py
-TestNdArray.py
-TestOutputFolder.py
-TestRelations.py
-TestSyntaxInterface.py
-TestVariousProtocols.py
+test_algebraic_models.py
```
Comment (jonc125) [inline test/ContinuousTestPack.txt]: We can probably get rid of these files, but perhaps leave that for another PR!

File: test/test_CompactSyntaxParser.py [hunk 20]
```diff
@@ -466,35 +465,35 @@ def testParsingOutputSpecifications(self):
 """, [[['n1', 'n2', 'u1'], ['n3', 'p:m', 'd1'], ['n4', 'u2', 'd2'], ['optional', 'n5', 'u3']]])
         self.assertParses(csp.outputs, "outputs {}", [[]])
 
-    def testParsingPlotSpecifications(self):
+    def test_parsing_plot_specifications(self):
         # TODO: Test these more once the XML syntax catches up!
```
Comment (jonc125) [inline test/test_CompactSyntaxParser.py]: Redundant todo ;)

File: .travis.yml [hunk 21]
```diff
@@ -13,7 +13,7 @@ branches:
 install:
   # Needed to run setup.py
   - pip install -e git://github.com/ModellingWebLab/cellmlmanip.git#egg=cellmlmanip
-  - pip install -e git://github.com/ModellingWebLab/weblab_cg#egg=weblab_cg
+  - pip install -e git://github.com/ModellingWebLab/weblab_cg@fc-refactoring#egg=weblab_cg
```
[the remark this answers is not in the document]
Comment (MichaelClerx) [reply .travis.yml]: Yes! Is that the sensible way to do it?

[the remark this answers is not in the document]
Comment (jonc125) [reply .travis.yml]: Could go either way. At present the tests on this branch will fail if it's changed. We could merge the weblab_cg branch first though, then change here and merge shortly after?
